imports/wasi_snapshot_preview1/wasi.go

```diff
@@ -274,12 +275,12 @@ func newHostFunc(
 
 // wasiFunc special cases that all WASI functions return a single Errno
 // result. The returned value will be written back to the stack at index zero.
-type wasiFunc func(ctx context.Context, mod api.Module, params []uint64) Errno
+type wasiFunc func(ctx context.Context, mod api.Module, params []uint64) syscall.Errno
 
 // Call implements the same method as documented on api.GoModuleFunction.
 func (f wasiFunc) Call(ctx context.Context, mod api.Module, stack []uint64) {
 	// Write the result back onto the stack
-	stack[0] = uint64(f(ctx, mod, stack))
+	stack[0] = uint64(ToErrno(f(ctx, mod, stack)))
```

Comment:
I would special case 0 here to avoid ToErrono as this is the all the WASI function's path and I guess most of the time  return zero

Reply:
great idea